### ISSUE TITLE
Ghost 0.7 Overview

### ISSUE THREAD

**Issue by ErisDS** [TryGhost/Ghost#5503]
This is a place for me to bring together an overview of all of the work that's currently ongoing towards the next release of Ghost, which will be 0.7. 

With Zelda, we're undertaking enormous changes to the design of Ghost's UI. At the same time we're hoping to land a number of new features and improvements which build on the freedom Zelda gives us and also leverage the major version bump as a hint things might be rocky during the upgrade this time!

There are 3 aspects to the changes that are going into 0.7, each covered in more detail below.

- Zelda
- Public API
- Data Migration

## Zelda 

Zelda is the UI redesign projects, which also includes a few major functional additions and changes:

- [x] autocomplete search box
- [x] new onboarding flow
- [x] overhaul of notifications, alerts, and inline validation
- [x] rebuilding the tag component

All of this work is tracked separately in #5314

## Public API

We've been talking for a very long time about making our JSON API publicly accessible via OAuth (See the [OAuth Epic](https://github.com/TryGhost/Ghost/issues/4004)).

The plan is to make this a reality in 0.7, but in a limited way: The read-only public endpoints will be made available to themes via the `{{#get}}` helper and also to public clients, but without a UI for registering new clients.

This means that it will be possible to fetch what we consider 'public' data - published posts, tags, and active users. This will be considered a 'public beta' of the API. 

The `{{#get}}` helper allows for accessing that data inside a theme, so that it is possible to fetch a tag list, or recent posts, etc. (Issue #4439)

JavaScript clients which run in the browser will also be able to access the data, providing they have a client id (See #4180). It will be possible to insert new client ids directly into the DB (a UI will come later). We will also make a default id available so that themes can easily make use of ajax to access the API (see #4184).

As well as making the data accessible, we need to push to clean up the API and ensure that the available endpoints are truly useful. The following are a set of must-have and nice-to-have API changes:

- [ ] Validation (issues #2756 & #2758) - in progress by @erisds
-  ~~Pagination with offset instead of page (#5093)~~
- [x] Make the 401 error format consistent (#5499)
- [ ] Add `fields` parameter (issue #5601)
- [ ] Add the `filter` parameter (issue #5604)
- [ ] Move top level browse params to `filter` (issue #5463)
- [ ] No more staticPages parameter + better pages support api (#5151, requires #5604)
- [ ] Add `order` parameter (issue #5602, ref #5463)
- [ ] Finish 'pipeline' refactor #5508 (hold until validation is complete)
- [ ] Joins & relations cleanup (issue #5615)

Handling permissions / auth:

- [ ] Public permission handling (#5614 partial PR in progress)
- [ ] Remove auth from public endpoints (#4181)
- [ ] Implement auth for public clients with a client id (#4180)
- [ ] Default client entry (#4184)

Last but not least, the get helper is currently a single item:

- [ ] Initial get helper (#4439)
- [ ] Update get helper to work with Filter param (Issue TBD)
- [ ] Improve the has helper (issue TBD)

Extras:

- [ ] Improve error messages 
- [ ] Figure out what to do with versioning

## Data Migration 

Ghost's internal migration system centres around changes to the database schema. For 0.7 there are a number of candidates for things which may change the schema. Once a schema change is initiated we can also hook in other migrations that don't affect the database schema directly, but require a database version change to be triggered.

Possible schema changes:

- [x] oAuth clients (#4174) - this is required to make public API access a reality
- [x] user tour (#5171) - an additional field for the user table, relatively straightforward and perhaps worth doing now
- [x] ~~post preview (#5097) - the spec for this is a little simplistic, so it may be worth doing this later when the spec has been reviewed and improved~~
- [x] tag order (issue TBD) - issue on this coming soon, would be good to get done

Other migrations that need to be added in:

- [x] jQuery removal (#5298) - this is a long awaited change, it would be preferable to get this done
- [x] Change private blog settings (issue #5614) - this is important for ensuring the API remains secure when public access is added

# Extras

- [x] Update top-gh-contribs to better reflect who contributed to a release
- [x] Refactor, clean up and improve test-coverage on Ghost's middleware #5286
- [ ] Prep work for channels (see [Channels 101](https://github.com/TryGhost/Ghost/wiki/Channels-101))

**Comment by sgoodwin** [comment]
Public API bits are a start. Yay API finally!

**Comment by ErisDS** [comment]
Once #5706 is merged, this work will be as complete as it is going to get for the first 0.7 release. Autocomplete, public API & get helper work will follow in a subsequent release - so punting this to the Post-Zelda milestone.

**Comment by ErisDS** [comment]
Closing in favour of #5976 which is tracking on going work
